diff --git a/source/rofi-icon-fetcher.c b/source/rofi-icon-fetcher.c
index fd9fd25b..3c74c5dd 100644
--- a/source/rofi-icon-fetcher.c
+++ b/source/rofi-icon-fetcher.c
@@ -217,7 +217,6 @@ rofi_icon_fetcher_get_surface_from_pixbuf(GdkPixbuf *pixbuf) {
   alpha = gdk_pixbuf_get_has_alpha(pixbuf);
 
   cairo_surface_t *surface = NULL;
-
   gint cstride;
   guint lo, o;
   guchar a = 0xff;
@@ -229,6 +228,7 @@ rofi_icon_fetcher_get_surface_from_pixbuf(GdkPixbuf *pixbuf) {
   lo = o * width;
 
   surface = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, width, height);
+  //cairo_surface_set_device_scale(surface, 2, 2);
   cpixels = cairo_image_surface_get_data(surface);
   cstride = cairo_image_surface_get_stride(surface);
 
@@ -321,7 +321,7 @@ static void rofi_icon_fetcher_worker(thread_state *sdata,
   } else {
     icon_path = icon_path_ = nk_xdg_theme_get_icon(
         rofi_icon_fetcher_data->xdg_context, themes, NULL, sentry->entry->name,
-        MIN(sentry->wsize, sentry->hsize), 1, TRUE);
+        MIN(sentry->wsize, sentry->hsize), 2, TRUE);
     if (icon_path_ == NULL) {
       g_debug("failed to get icon %s(%dx%d): n/a", sentry->entry->name,
               sentry->wsize, sentry->hsize);
@@ -347,7 +347,7 @@ static void rofi_icon_fetcher_worker(thread_state *sdata,
 
   GError *error = NULL;
   GdkPixbuf *pb = gdk_pixbuf_new_from_file_at_scale(
-      icon_path, sentry->wsize, sentry->hsize, TRUE, &error);
+      icon_path, sentry->wsize*2, sentry->hsize*2, TRUE, &error);
   if (error != NULL) {
     g_warning("Failed to load image: %s", error->message);
     g_error_free(error);
@@ -446,3 +446,4 @@ cairo_surface_t *rofi_icon_fetcher_get(const uint32_t uid) {
   }
   return NULL;
 }
+
