From 93b2a1ad33bcf21fc4eb1713c19ac9b290cd0a62 Mon Sep 17 00:00:00 2001
From: lbonn <github@lbonnans.net>
Date: Fri, 10 Feb 2023 22:39:05 +0100
Subject: [PATCH] Fix redraw bug when changing window height

See #59
---
 source/wayland/view.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/wayland/view.c b/source/wayland/view.c
index 1ace96630..8e3431c22 100644
--- a/source/wayland/view.c
+++ b/source/wayland/view.c
@@ -155,6 +155,7 @@ static void wayland_rofi_view_window_update_size(RofiViewState *state) {
   display_set_surface_dimensions(state->width, state->height,
                                  offset_x, offset_y,
                                  rofi_get_location(state));
+  rofi_view_pool_refresh();
 }
 
 static void wayland_rofi_view_set_size(RofiViewState *state, gint width,
@@ -402,8 +403,7 @@ static int wayland_rofi_view_calculate_window_height(RofiViewState *state) {
   // Autosize based on widgets.
   widget *main_window = WIDGET(state->main_window);
 
-  height = widget_get_desired_height(main_window, state->width);
-  return height;
+  return widget_get_desired_height(main_window, state->width);
 }
 
 static void wayland_rofi_view_hide(void) {}

